### DISCUSSION TITLE
Plans to support the "New MS Teams"?

### DISCUSSION THREAD

**Discussion by taeuschkoerper** [Claus-E/teams2tasmota#8] [Q&A]
Hi Claus,
First of all, thank you for that nice piece of Software. Unfortunately we switched at work to the new teams and it is now not working anymore. Are there plans to support the new MS Teams?
And somehow I am to stupid to get it work via USB. I am using a D1 mini ESP8266 with tasmota 13.4 and a WS2812b LED ring. The USB/serial connection is working. Any ideas? Thank you for your support and best regards from Hamburg

**Comment by Claus-E** [comment]
Hi tauschkoerper,

I have the same problem at work, the "new Teams".
MS changed many parts for the new Teams and unfortunately the logfile which was the base for my teams2tasmota doesn’t exist anymore.
But I found a new logfile with a completely new layout. I am trying to adapt it, but it will not have the same behavior.
I am trying to adapt the teams2tasmota to the new logfile.

About your USB/serial connection. Did you tested the connection with a terminal program like hterm (https://www.heise.de/download/product/hterm-53283)?
Select the comport and the Baudrate 115200,8,1, none
If you send a command like “Color FF0000” + LF you will get an answer from tasmota.
If this works, then teams2tasmota should work also. If not you can try to change the baudrate, but I think 115200 is the default baudrate.
Easter greetings from Lower Saxony
Claus

**Comment by taeuschkoerper** [comment]
Guten Morgen Claus,

Danke für die schnelle Antwort und den Tipp. Da hätte ich auch Mal selbst drauf kommen können🫣.

Dann drücke ich die Daumen, dass das mit dem neuen Logfile klappt.
Die anderen Umsetzungen über die Graph API finde ich auch völlig überkandidelt 😁.

Besten Dank und schöne Ostertage.

Armin

**Reply by Claus-E** [reply]
Hi Armin,

die neue Version 0.5.0 von teams2tasmota mit Support für 'new Teams' ist online. Ich habe noch nicht viel getestet, wird sicher noch Bugs enthalten. Wenn Du welche findest, kannst Du einfach hier schreiben.

Schönen Abend noch, Gruß Claus

**Comment by taeuschkoerper** [comment]
Hallo Claus,
besten Dank. Schaue ich mir Mittwoch Abend gleich an. Und es war natürlich die Baudrate, warum es bei mir nicht ging.
Ich wünsche ebenfalls einen schönen Abend.
Schöne Grüße
Armin

**Comment by taeuschkoerper** [comment]
Hallo Claus,

bis lang funktioniert alles. Zu Beginn machte es den Eindruck, als würde es einen Moment brauchen, bis die Software die Einstellungen aus Teams übernimmt, aber dann scheint es stabil zu laufen.
Sehr gut finde ich, dass der Pfad zur LOG Datei in der XML angepasst werden kann.
Jetzt muss ich nur noch ein passendes Gehäuse konstruieren für den LED Ring mit dem D1 mini und dann den Drucker anschmeißen 🤗.
Vielen Dank und einen schönen Abend
Armin

01.04.2024 22:08:52 Armin Mertens ***@***.***>:

**Reply by h07d0q** [reply]
Oh, an dem 3D Modell wäre ich sehr interessiert 😊
Würdest du es allen zur Verfügung stellen?

**Reply by taeuschkoerper** [reply]
Hi,
Und hier die 3D Files. Einmal die *.stl und die Freecad Dateien. Da ich zu faul war etwas Aufwendiges zu konstruieren, ist der Entwurf so einfach wie möglich geworden. Die beiden Teile lassen sich ineinander stecken und halten von selbst.
Am Ende braucht es nur etwas Kleber für den LED Ring und den D1 mini. Und etwas Klettband für den Monitor oder wo es auch immer festgemacht werden soll. 
[Teams Light.zip](https://github.com/Claus-E/teams2tasmota/files/14896668/Teams.Light.zip)

Anbei noch ein paar Bilder.
![20240407_113032](https://github.com/Claus-E/teams2tasmota/assets/119073969/470cd7a8-98c0-45a8-bfbd-3a870d793d7f)
![20240407_113109](https://github.com/Claus-E/teams2tasmota/assets/119073969/d6679ee6-5b52-4d88-8171-3c82d51e02e8)
![20240407_113021](https://github.com/Claus-E/teams2tasmota/assets/119073969/e5a47dd9-ff95-4649-9fd7-e90d990c735c)
![20240407_113143](https://github.com/Claus-E/teams2tasmota/assets/119073969/d35e0c42-0884-4630-a533-658d82a201c9)
Und natürlich habe ich den Ausschnitt im Deckel vergessen, weswegen das auf dem Bild mit dem Lötkolben raus geschnitten ist :-P.
In den 3D Dateien ist das aber angepasst.

Hier noch die Links zu den Komponenten.
https://www.az-delivery.de/products/kopie-von-rgb-led-ring-ws2812-mit-12-rgb-leds-5v-fuer-arduino
https://www.az-delivery.de/products/d1-mini
Und der Kabelplan
![Teams Light Kabelplan](https://github.com/Claus-E/teams2tasmota/assets/119073969/48574b86-49a2-483b-82cd-49e7c01ce09a)

Jetzt muss @Claus-E  nur noch hinbekommen, dass das mit dem neuen Teams vollständig läuft :-).

Schönen Sonntag noch.

**Reply by h07d0q** [reply]
Cool, vielen dank für die Dateien und den Aufwand! 
Werde es als Inspiration für meine 8 Pixel LED Streifen nehmen. 
Wie erreichst du den USB Port mit dem Kabel? Geht der ganze Stecker durch das Gehäuse, oder hast du am Rand eine zweite Buchse?

**Reply by JStaegerino** [reply]
Vielen Dank @taeuschkoerper 
Scheit mir mit dem D1-Mini wesentlich einfacher und auch günstiger zu sein.
Habe mir soeben die Komponenten bestellt und werde ich in den Festtagen mal umsetzen 😀

**Comment by taeuschkoerper** [comment]
Das mache ich sobald ich was vernünftiges gemacht habe 😊 

 

Von: h07d0q ***@***.***> 
Gesendet: Mittwoch, 3. April 2024 20:47
An: Claus-E/teams2tasmota ***@***.***>
Cc: taeuschkoerper ***@***.***>; Author ***@***.***>
Betreff: Re: [Claus-E/teams2tasmota] Plans to support the "New MS Teams"? (Discussion #8)

 

Oh, an dem 3D Modell wäre ich sehr interessiert 😊
Würdest du es allen zur Verfügung stellen?

**Comment by taeuschkoerper** [comment]
Guten Morgen Claus,

Ich konnte jetzt nochmal etwas intensiver testen.
Die auswählbaren Status wie "busy" oder "away" funktionieren soweit. Was nicht geht, sind die Status wie "on the phone" oder "presentation".
Schöne Grüße

Armin

03.04.2024 19:59:19 Armin Mertens ***@***.***>:

**Comment by Claus-E** [comment]
Hi Armin,

Stimmt,das hatte ich auch schon beobachtet. In der Neuen Log-Datei wird das nicht unterschieden. Stattdessen wird Busy bzw. DoNotDisturb gesetzt 

Was ich noch nicht testen konnte ist, was passiert, wenn ein Anruf eingeht. Ich befürchte da gibt es nichts worauf ich reagieren kann.

Nächste Woche muss ich wieder arbeiten und somit zwangsläufig testen 😁

Gruß Claus

**Comment by taeuschkoerper** [comment]
Moin,
Die Aussparung in der Seite ist groß genug für den ganzen USB Stecker.
Das Loch ist 15mm breit und 10mm hoch.
Schöne Grüße

07.04.2024 22:18:39 h07d0q ***@***.***>:
